src/main/java/net/minecraftforge/network/HandshakeHandler.java

```diff
@@ -186,13 +195,6 @@ void handleServerModListOnClient(HandshakeMessages.S2CModList serverModList, Sup
         c.get().setPacketHandled(true);
         //The connection data needs to be modified before a new ModMismatchData instance could be constructed
         NetworkHooks.appendConnectionData(c.get().getNetworkManager(), serverModList.getModList().stream().collect(Collectors.toMap(Function.identity(), s -> Pair.of("", ""))), serverModList.getChannels());
-        if (!mismatchedChannels.isEmpty()) {
-            LOGGER.error(FMLHSMARKER, "Terminating connection with server, mismatched mod list");
-            //Populate the mod mismatch attribute with a new mismatch data instance to indicate that the disconnect happened due to a mod mismatch
-            c.get().getNetworkManager().channel().attr(NetworkConstants.FML_MOD_MISMATCH_DATA).set(ModMismatchData.channel(mismatchedChannels, NetworkHooks.getConnectionData(c.get().getNetworkManager()), true));
-            c.get().getNetworkManager().disconnect(Component.literal("Connection closed - mismatched mod channel list"));
-            return;
```

Comment:
Removing this disconnect prevents the connection from aborting if the mod list is mismatched, but there is another check below which will fail if the datapack registries are mismatched (which is likely to happen if the mod list is mismatched).

A more proper solution is to remove both disconnects from this handler, and instead allow the server to manage the disconnects via a reply message. This could be baked into `C2SModListReply` with an additional "clientChecksPassed" boolean _or_ a new packet specifically for allowing the server to determine the status.  This point would then be the appropriate time to reply with `S2CReset`, or aborting the connection if a retry is not necessary.